services/network/client/infrastructure/input/tcp-check.controller.go

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2022 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+package input
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	pb "github.com/Snowflake-Labs/sansshell/services/network"
+	app "github.com/Snowflake-Labs/sansshell/services/network/client/application"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+	cliUtils "github.com/Snowflake-Labs/sansshell/services/util/cli"
+	"github.com/Snowflake-Labs/sansshell/services/util/validator"
+	"github.com/google/subcommands"
+	"os"
+)
+
+// TCPCheckCmd cli adapter for execution infrastructure implementation of [subcommands.Command] interface
+type TCPCheckCmd struct {
+	host    string
+	port    int
+	timeout uint
+}
+
+func (*TCPCheckCmd) Name() string { return "tcp-check" }
+func (*TCPCheckCmd) Synopsis() string {
+	return "Check tcp connectivity from remote machine to specified server"
+}
+func (*TCPCheckCmd) Usage() string {
+	return `tcp-check --host <host> --port <port>:
+    Makes tcp connectivity check from the remote machine to specified server.
+`
+}
+
+func (p *TCPCheckCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&p.host, "host", "", "Host to check connectivity from remote machine")
+	f.IntVar(&p.port, "port", 8080, "Port to check connectivity from remote machine")
+	f.UintVar(&p.timeout, "timeout", 3, "Timeout in seconds to wait for response from --host on remote machine")
+}
+
+// Execute is a method handle command execution. It adapter between cli and business logic
+func (p *TCPCheckCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	state := args[0].(*util.ExecuteState)
+	if p.host == "" {
+		cliUtils.ColoredFprintf(os.Stderr, cliUtils.RED_TEXT_COLOR, "--host flag is required.\n")
+		return subcommands.ExitUsageError
+	}
+
+	if err := validator.IsValidPort(p.port); err != nil {
+		cliUtils.ColoredFprintf(os.Stderr, cliUtils.RED_TEXT_COLOR, "Invalid port number: %s\n", err.Error())
+		return subcommands.ExitUsageError
+	}
+
+	preloader := cliUtils.NewDotPreloader("Waiting for results from remote machines")
```

Comment:
This part feels like it should be in sanssh-level code that applies to all subcommands and not in an individual action. Maybe you can do a separate PR that does it?

Reply:
Not sure. Controller prints result from each host as it get results and of keeps this preloaded very last line until all targets not gives its reply. IMAO, how to handle progress-bar/pre-loader it hardly depends on what commands do on local machine and how it handles server outputs, extract this logic on a sanssh-level could be either tricky or may not work in all cases

Reply:
I think uniformity across commands is useful. I don't mind changing how our outputting works, but I'd rather do it for all commands and not a subset of them.

Reply:
Let me become more familiar with a project and I'll be able to refactor it.